src/map/unit.cpp

```diff
@@ -1831,16 +1833,35 @@ void unit_set_attackdelay(block_list& bl, t_tick tick)
 
 	switch (bl.type) {
 		case BL_PC:
-			ud->attackabletime = tick + status_get_adelay(&bl);
-			// A fixed delay is added here which is equal to the minimum attack motion you can get
-			// This ensures that at max ASPD attackabletime and canact_tick are equal
-			ud->canact_tick = tick + status_get_amotion(&bl) + (pc_maxaspd(reinterpret_cast<map_session_data*>(&bl)) / AMOTION_DIVIDER_PC);
+			switch (event) {
+				case DELAY_EVENT_ATTACK:
+				//case DELAY_EVENT_CASTBEGIN_ID:
+				//case DELAY_EVENT_CASTBEGIN_POS:
+				case DELAY_EVENT_PARRY:
+					ud->attackabletime = tick + status_get_adelay(&bl);
+					// A fixed delay is added here which is equal to the minimum attack motion you can get
+					// This ensures that at max ASPD attackabletime and canact_tick are equal
+					ud->canact_tick = tick + status_get_amotion(&bl) + (pc_maxaspd(reinterpret_cast<map_session_data*>(&bl)) / AMOTION_DIVIDER_PC);
+					break;
+			}
 			break;
 		case BL_MER:
-			// TODO: Should set this, but only for ground skills
+			switch (event) {
+				case DELAY_EVENT_ATTACK:
+					ud->attackabletime = tick + status_get_adelay(&bl);
+					break;
+				//case DELAY_EVENT_CASTBEGIN_POS:
+				//	ud->attackabletime = tick + status_get_adelay(&bl);
+				//	ud->canact_tick = tick + status_get_amotion(&bl) + MAX_ASPD_NOPC;
+				//	break;
```

Comment:
```suggestion
```
Should be added, when it is time.
Which leaves BL_MER = default case for the time being...